test/v2/generators/DepartureMessageGeneratorSpec.scala

```diff
@@ -0,0 +1,89 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.generators
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import v2.models.DepartureId
+import v2.models.MessageType
+import v2.models.MessageType.MRNAllocated
+
+import java.time.Clock
+import scala.xml.NodeSeq
+
+class DepartureMessageGeneratorSpec extends AnyFreeSpec with Matchers {
+
+  import Setup._
+  "A generator" - {
+    "when supplied with message type MRNAllocated" - {
+      "should produce an IE028 Message" - {
+
+        val traderChannelResponse = messages(MRNAllocated)
+
+        // header
+        val messageSender          = (traderChannelResponse \ "CC028C" \ "messageSender").text
+        val messageRecipient       = (traderChannelResponse \ "CC028C" \ "messageRecipient").text
+        val preparationDateAndTime = (traderChannelResponse \ "CC028C" \ "preparationDateAndTime").text
+        val messageIdentification  = (traderChannelResponse \ "CC028C" \ "messageIdentification").text
+        val messageType            = (traderChannelResponse \ "CC028C" \ "messageType").text
+        val correlationIdentifier  = (traderChannelResponse \ "CC028C" \ "correlationIdentifier").text
+
+        // TransitOperation
+        val mrn                       = (traderChannelResponse \ "CC028C" \ "TransitOperation" \ "MRN").text
+        val declarationAcceptanceDate = (traderChannelResponse \ "CC028C" \ "TransitOperation" \ "declarationAcceptanceDate").text
+
+        // CustomsOfficeOfDeparture
+        val referenceNumber = (traderChannelResponse \ "CC028C" \ "CustomsOfficeOfDeparture" \ "referenceNumber").text
+
+        // Header
+        messageSender should fullyMatch regex messageSenderPattern
+        messageRecipient should fullyMatch regex messageRecipientPattern
+        preparationDateAndTime should fullyMatch regex preparationDateAndTimePattern
+        messageIdentification should fullyMatch regex messageIdentificationPattern
+        messageType should fullyMatch regex messageTypePattern
+        correlationIdentifier should fullyMatch regex correlationIdentifierPattern
+
+        // TransitOperation
+        messages.isDefinedAt(MRNAllocated) should be(true)
+        mrn should fullyMatch regex mrnPattern
+        declarationAcceptanceDate should fullyMatch regex datePattern
+
+        // CustomsOfficeOfDeparture
+        referenceNumber should fullyMatch regex referenceNumberPattern
+
+      }
+    }
+  }
+
+  object Setup {
+    val clock                                           = Clock.systemUTC()
+    val ut                                              = new DepartureMessageGenerator(clock)
+    val messages: PartialFunction[MessageType, NodeSeq] = ut.generate(DepartureId("departureId"))
```

Comment:
I'd rather we generated a 16 hex string and tested against that, rather than just a string like this, then check that the regex holds with the 16 zeroes.

Reply:
Not sure it's relevant but will change.